share/wake/lib/core/integer.wake

```diff
@@ -13,78 +13,80 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
 package wake
 
 # The Integer type has unbounded precision (aka a "big" integer).
 # Any wake literal like 12331_1232 or 0x3123 is an Integer.
 
+
 # Unary positive sign operator for Integer values.
 # (+5) = 5
-export def +(x: Integer): Integer =
-    x
+export def +(x: Integer): Integer = x
+
 
 # Unary negative sign operator for Integer values.
 # (-5) = 0-5
-export def -(x: Integer): Integer =
-    (\_ prim "neg") x
+export def -(x: Integer): Integer = (\_ prim "neg") x
+
 
 # Unary two's complement operator for Integer values.
 # ~0 = -1
 # ~4 = -5
-export def ~(x: Integer): Integer =
-    (\_ prim "com") x
+export def ~(x: Integer): Integer = (\_ prim "com") x
+
 
 # Binary addition operator for Integer values.
 # 1 + 2 = 3
 # 1 + 5 = 6
-export def (x: Integer) + (y: Integer): Integer =
-    (\_\_ prim "add") x y
+export def (x: Integer) + (y: Integer): Integer = (\_\_ prim "add") x y
+
 
 # Binary subtraction operator for Integer values.
 # 2 - 1 =  1
 # 3 - 4 = -1
-export def (x: Integer) - (y: Integer): Integer =
-    (\_\_ prim "sub") x y
+export def (x: Integer) - (y: Integer): Integer = (\_\_ prim "sub") x y
+
 
 # Binary multiplication operator for Integer values.
 #  3 *   4  = 12
 # -3 * (-4) = 12
-export def (x: Integer) * (y: Integer): Integer =
-    (\_\_ prim "mul") x y
+export def (x: Integer) * (y: Integer): Integer = (\_\_ prim "mul") x y
+
 
 # Binary division operator for Integer values.
 # 12 / 3 =  4
 # 13 / 3 =  4
 # -8 / 4 = -2
-export def (x: Integer) / (y: Integer): Integer =
-    (\_\_ prim "div") x y
+export def (x: Integer) / (y: Integer): Integer = (\_\_ prim "div") x y
+
```

Comment:
Incoming bikeshed. 🚲 🚳 Haha.

I think I would prefer the formatter to not unformat newlines, particularly in cases like this (i.e., function definitions). The Elixir formatter *generally* abides by this, but the F# formatter (called Fantomas) does not, where I believe it is contentious.

For a simple function, it's probably okay, but I prefer the function, arguments, and type annotations line to be just that except for very simple functions or bindings. I probably have broken my own rule several times (although if I was code reviewing myself, I'd probably ask myself to actually follow the rule), but when I don't, I think I'd prefer the formatter not to break it for me.

I'm thinking my rubric is:

* If a top-level function definition, leave function name, arguments, and type annotations on a line by themselves, with the function expression starting on a newline. I.e.,
    ```
    def <functionName> <args> : <returnType> =
        <expression>
    ```
* If the top-level function body expression is *very* simple, allow it on the same line.
* For local function and value definitions, leave it to the discretion of the writer, particularly since these aren't as likely to have full type annotations.

Reply:
> If a top-level function definition, leave function name, arguments, and type annotations on a line by themselves, with the function expression starting on a newline. I.e., 

Rules just for top-level are not super trivial (at least until we add some kind of "you are top level" signal) and I want to punt on adding them. If we do this for _all_ defs then that is quite easy to do.

> If the top-level function body expression is very simple, allow it on the same line.

Can you give a formal definition for "very simple" :)

>  For local function and value definitions, leave it to the discretion of the writer, particularly since these aren't as likely to have full type annotations.

I disagree. We should be opinionated, otherwise there isn't a point in having the formatter. The more discretion we give the more inconsistency the codebase has and the more opportunity for formatting bikesheds in random PRs